From 486a17f8a1cc8290f1fb0f2f45b6f624d15e637b Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Mon, 29 Aug 2016 15:55:38 +0000
Subject: [PATCH 2499/2848] HADOOP-13552. RetryInvocationHandler logs all
 remote exceptions. Contributed by Jason Lowe
 (cherry picked from commit
 92d8f371553b88e5b3a9d3354e93f75d60d81368)

(cherry picked from commit 039c831f5f7b1668c7487c89e12d482ede08fbbe)

 Conflicts:
	hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java

Change-Id: Iccab3456bbb4120389503e6d100a09526554a343
---
 .../hadoop/io/retry/RetryInvocationHandler.java    |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java
index d57dc84..dd56882 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/retry/RetryInvocationHandler.java
@@ -123,9 +123,11 @@ public Object invoke(Object proxy, Method method, Object[] args)
         if (failAction != null) {
           // fail.
           if (failAction.reason != null) {
-            LOG.warn("Exception while invoking " + currentProxy.proxy.getClass()
-                + "." + method.getName() + " over " + currentProxy.proxyInfo
-                + ". Not retrying because " + failAction.reason, ex);
+            if (LOG.isDebugEnabled()) {
+              LOG.debug("Exception while invoking " + currentProxy.proxy.getClass()
+                  + "." + method.getName() + " over " + currentProxy.proxyInfo
+                  + ". Not retrying because " + failAction.reason, ex);
+            }
           }
           throw ex;
         } else { // retry or failover
-- 
1.7.9.5

